### PR TITLE
DB support

### PR DESCRIPTION
**Pull request by Amelia0010**
honk

### CODE REVIEW

**Comment by Amelia0010** [comment]
oh and this is what I worked from https://github.com/yogstation13/yogstation/commit/dca81c68b10d94ed4d254e2d99ed9f7ff9f66b7e#diff-6f61adba58272a6ded4c7387effcac4e

File: yogstation/code/modules/admin/admin_ranks.dm [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
...lity to ban, jobban and fullban\n# +STEALTH = the ability to stealthmin (make yourself appear with a fake name to everyone but other admins\n# +POSSESS = the ability to possess objects\n# +REJUV (or +REJUVINATE) = the ability to heal, respawn, modify damage and use godmode\n# +BUILD (or +BUILDMODE) = the ability to use buildmode\n# +SERVER = higher-risk admin verbs and abilities, such as those which affect the server configuration.\n# +DEBUG = debug tools used for diagnosing and fixing problems. It's useful to give this to coders so they can investigate problems on a live server.\n# +VAREDIT = everyone may view viewvars/debugvars/whatever you call it. This keyword allows you to actually EDIT those variables.\n# +RIGHTS (or +PERMISSIONS) = allows you to promote and/or demote people.\n# +SOUND (or +SOUNDS) = allows you to upload and play sounds\n# +SPAWN (or +CREATE) = mob transformations, spawning of most atoms including mobs (high-risk atoms, e.g. blackholes, will require the +FUN flag too)\n# +EVERYTHING (or +HOST or +ALL) = Simply gives you everything without having to type every flag\n\n# DO NOT EDIT THIS FILE DIRECTLY\n# IT IS AUTOMATICALLY GENERATED FROM THE SERVER DATABASE\n"
```
Comment (ThatLing) [inline yogstation/code/modules/admin/admin_ranks.dm]: Use WRITE_FILE

File: yogstation/code/modules/admin/admin_ranks.dm [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		if(query_admin.Execute())
+			fdel("config/admins.txt")
+			var/adminsFile = file("config/admins.txt")
+			adminsFile << "###############################################################################################\n# Basically, ckey goes first. Rank goes after the '='                                         #\n# Case is not important for ckey.                                                             #\n# Case IS important for the rank.                                                             #\n# All punctuation (spaces etc) EXCEPT '-', '_' and '@' will be stripped from rank names.      #\n# Ranks can be anything defined in admin_ranks.txt                                            #\n# NOTE: if the rank-name cannot be found in admin_ranks.txt, they will not be adminned! ~Carn #\n# NOTE: syntax was changed to allow hyphenation of ranknames, since spaces are stripped.      #\n###############################################################################################\n\n# DO NOT EDIT THIS FILE DIRECTLY\n# IT IS AUTOMATICALLY GENERATED FROM THE SERVER DATABASE\n"
```
Comment (ThatLing) [inline yogstation/code/modules/admin/admin_ranks.dm]: Use WRITE_FILE

File: code/modules/admin/DB_ban/functions.dm [hunk 3]
```diff
@@ -114,7 +114,7 @@
 	reason = sanitizeSQL(reason)
 
 	if(maxadminbancheck)
-		var/datum/DBQuery/query_check_adminban_amt = SSdbcore.NewQuery("SELECT count(id) AS num FROM [format_table_name("ban")] WHERE (a_ckey = '[a_ckey]') AND (bantype = 'ADMIN_PERMABAN'  OR (bantype = 'ADMIN_TEMPBAN' AND expiration_time > Now())) AND isnull(unbanned)")
+		var/datum/DBQuery/query_check_adminban_amt = SSdbcore.NewQuery("SELECT count(id) AS num FROM [format_table_name("ban")] WHERE (a_ckey = '[a_ckey]') AND (bantype = 'ADMIN_PERMABAN'  OR (bantype = 'ADMIN_TEMPBAN' AND expiration_time > Now())) AND unbanned == 0") //yogs - DB support
```
Comment (ThatLing) [inline code/modules/admin/DB_ban/functions.dm]: delete this change

File: code/modules/admin/IsBanned.dm [hunk 4]
```diff
@@ -70,15 +70,15 @@
...
+		var/cidquery = "" YOGS - DB support
 		if(address)
 			ipquery = " OR ip = INET_ATON('[address]') "
 
 		if(computer_id)
-			cidquery = " OR computerid = '[computer_id]' "
+			cidquery = " OR computerid = '[computer_id]' " */ //Yogs end - DB support
 
-		var/datum/DBQuery/query_ban_check = SSdbcore.NewQuery("SELECT ckey, a_ckey, reason, expiration_time, duration, bantime, bantype, id, round_id FROM [format_table_name("ban")] WHERE (ckey = '[ckeytext]' [ipquery] [cidquery]) AND (bantype = 'PERMABAN' OR bantype = 'ADMIN_PERMABAN' OR ((bantype = 'TEMPBAN' OR bantype = 'ADMIN_TEMPBAN') AND expiration_time > Now())) AND isnull(unbanned)")
+		var/datum/DBQuery/query_ban_check = SSdbcore.NewQuery("SELECT ckey, a_ckey, reason, expiration_time, duration, bantime, bantype, id, round_id FROM [format_table_name("ban")] WHERE (ckey = '[ckeytext]') AND (bantype = 'ADMIN_PERMABAN' OR (bantype = 'ADMIN_TEMPBAN' AND expiration_time > Now())) AND unbanned == 0") //yogs - DB support
```
Comment (ThatLing) [inline code/modules/admin/IsBanned.dm]: delete this change

File: code/modules/admin/IsBanned.dm [hunk 5]
```diff
@@ -70,15 +70,15 @@
 			message_admins(msg)
 			return
 
-		var/ipquery = ""
-		var/cidquery = ""
+		/*var/ipquery = "" YOGS - DB support //yogs start - DB support
```
Comment (ThatLing) [inline code/modules/admin/IsBanned.dm]: delete this change

File: code/modules/admin/banjob.dm [hunk 6]
```diff
@@ -4,7 +4,7 @@
 		return FALSE
 
 	if(!M.client) //no cache. fallback to a datum/DBQuery
-		var/datum/DBQuery/query_jobban_check_ban = SSdbcore.NewQuery("SELECT reason FROM [format_table_name("ban")] WHERE ckey = '[sanitizeSQL(M.ckey)]' AND (bantype = 'JOB_PERMABAN'  OR (bantype = 'JOB_TEMPBAN' AND expiration_time > Now())) AND isnull(unbanned) AND job = '[sanitizeSQL(rank)]'")
+		var/datum/DBQuery/query_jobban_check_ban = SSdbcore.NewQuery("SELECT reason FROM [format_table_name("ban")] WHERE ckey = '[sanitizeSQL(M.ckey)]' AND (bantype = 'JOB_PERMABAN'  OR (bantype = 'JOB_TEMPBAN' AND expiration_time > Now())) AND unbanned == 0 AND job = '[sanitizeSQL(rank)]'") //yogs - DB support
```
Comment (ThatLing) [inline code/modules/admin/banjob.dm]: delete this change

File: code/modules/admin/banjob.dm [hunk 7]
```diff
@@ -26,7 +26,7 @@
 		return
 	if(C && istype(C))
 		C.jobbancache = list()
-		var/datum/DBQuery/query_jobban_build_cache = SSdbcore.NewQuery("SELECT job, reason FROM [format_table_name("ban")] WHERE ckey = '[sanitizeSQL(C.ckey)]' AND (bantype = 'JOB_PERMABAN'  OR (bantype = 'JOB_TEMPBAN' AND expiration_time > Now())) AND isnull(unbanned)")
+		var/datum/DBQuery/query_jobban_build_cache = SSdbcore.NewQuery("SELECT job, reason FROM [format_table_name("ban")] WHERE ckey = '[sanitizeSQL(C.ckey)]' AND (bantype = 'JOB_PERMABAN'  OR (bantype = 'JOB_TEMPBAN' AND expiration_time > Now())) AND unbanned == 0") //yogs - DB support
```
Comment (ThatLing) [inline code/modules/admin/banjob.dm]: delete this change

File: code/modules/admin/IsBanned.dm (hunk 5 above)
Comment (ThatLing) [reply code/modules/admin/IsBanned.dm]: these changes even

**Review by ThatLing — CHANGES_REQUESTED** [review]

**Review by ThatLing — APPROVED** [review]

**Pull request merged**